### ISSUE TITLE
CPU temperature goes to off state

### ISSUE THREAD

**Issue by tairosonloa** [kotelnik/plasma-applet-thermal-monitor#53]
I use this applet to show my CPU, GPU, SSD and HDD temperature. However, I have a problem with the CPU temperature.

When I start the computer, all the temperatures show its values except CPU, which shows an "off" message. Also, when I'm using my computer, sometimes I realize CPU temperature is "off" again. This can be manually fixed by right clicking on the applet and left click on "Reload temperature sources".

My CPU is a AMD Ryzen 5 1500x and I'm using the lmsensors/nct6795-isa-0a20/CPUTIN

I think this can be "patched" if a new feature to auto reload temperature sources on these situations is added.

**Comment by alium** [comment]
same problem with intel

**Comment by tairosonloa** [comment]
Also, I saw a day ago an user (Nano-Ocelot) wrote a comment on the kde store. I copy and paste:

> I have a weird bug where I have to 'refresh sources' after logon to display cpu core group temperatures (highest temp core shown). Otherwise it remains blank. However gpu temp (single temp source) works correctly on initial widget load.

To see the original comment:  [https://store.kde.org/p/998915/#ratings-panel](https://store.kde.org/p/998915/#ratings-panel)

**Comment by GuerreroNinja** [comment]
Same problem here with Manjaro. Seems something updated and broke the initial temp readings of the widget with CPU. I am on Intel btw.
EDIT: Seems this is the same issue as #16

**Comment by ademcal** [comment]
There is same problem I am using Manjaro on Dell Vostro 5568.

**Comment by emilio-aburto** [comment]
I'm also seeing this issue, Intel CPU and AntergOS. Seems like the problem was caused by some update in archlinux, but really don't know.

**Comment by lvxejay** [comment]
Issue still persists on Arch Linux-LTS Kernel v4.19.14-1, Intel i7 5820k CPU

**Comment by cryzed** [comment]
Same issue here, same kernel and OS as @lvxejay, CPU is Intel i7-7770K.

**Comment by roachsinai** [comment]
Same issue, Manjaro KDE, cup fan shows `OFF` also.

**Comment by afettouhi** [comment]
Can confirm this on Arch Linux as well since updating to the latest major KDE release 5.14. Using the latest git of plasma-applet-thermal-monitor btw.

**Comment by severnaside** [comment]
Same problem, but I found a solution (of sorts).  I installed Plasma 5 'Simple System Monitor' which sits on the desktop and I have Thermal Monitor in the panel.  Reboot - now Thermal Monitor has the CPU temperature loaded!  How this works I do not know, but maybe there's a clue for someone smarter than me to get Thermal Monitor showing CPU temperature at startup, by itself.
 Arch Linux, LTS kernel 4.19.15-1, Intel i7-7500U

**Comment by roachsinai** [comment]
@severnaside you mean this [plasmoid](https://store.kde.org/p/1173509/)?

**Comment by severnaside** [comment]
@roachsinai Yes

**Comment by roachsinai** [comment]
I think this issue #52 is useful, cause I comment https://github.com/kotelnik/plasma-applet-thermal-monitor/blob/783536952c38bf26aefadb36d2dadd50c8ddc2c0/package/contents/ui/main.qml#L312 thisl line. No need refresh by yourself. But it's not a so good method.

Cause I don't konw how to debug a plasmoid and have not learn javascript(just know a very little), someone maybe could debug the `QML type error`.

**Comment by roachsinai** [comment]
Maybe solve the problem as I found var `systemmonitorAvailableSources` should be an array, change the code according to [var QML Basic Type](http://doc.qt.io/qt-5/qml-var.html)... Made a pull request #54 , check it and update if you have a better solution.

**Comment by alium** [comment]
pull #54 fixed problem with manual reload not, i tested the patch today. After restart the computer came again OFF

**Comment by roachsinai** [comment]
@alium What's your Linux Distro? I only test it on `Manjaro`. For me there are 3 way to let cpu `OFF` disappear.

1. comment `systemmonitorAvailableSources.push(source)` at `plasma-applet-thermal-monitor/package/contents/ui/main.qml` line 312
2. change `property var systemmonitorAvailableSources` to `property var systemmonitorAvailableSources: []` at `plasma-applet-thermal-monitor/package/contents/ui/main.qml` line 58
3. pull #54 

So, I think maybe you could try others.

And make sure after you pull #54 , you also update code at `/usr/share/plasma/plasmoids/org.kde.thermalMonitor
`(maybe some difference according to Distro). Or just update manually.

So sorry that only you reply `Not Work`, haven't get a `Work` reply(but 2 thumb up at my previous reply).

**Comment by alium** [comment]
@roachsinai 
i try your instructions, but ended with error: file:///usr/share/plasma/plasmoids/org.kde.thermalMonitor/contents/ui/main.qml:312:1: Expected token `}'

seems as problem with comment "systemmonitorAvailableSources.push(source)"

use Archlinux with OpenRC (also Artix Linux)

**Comment by roachsinai** [comment]
First，those 3 method is mutually exclusive.

And for now I think the different between your sittuation an my is I installed systemed version Manjaro. So, I can't reproduce it. And I think the error is wierd.

Do you get the error message on booting?

**Comment by roachsinai** [comment]
What I did is try to solve #52 which let  got the idea that `systemmonitorAvailableSources` maybe not support push method. But maybe this is something different in problem, though change `systemmonitorAvailableSources` to an array is work for me.

**Comment by alium** [comment]
@roachsinai 
oh, ok! now i understand :-)  i try all 3 methods

**Comment by alium** [comment]
> 1.    comment systemmonitorAvailableSources.push(source) at plasma-applet-thermal-monitor/package/contents/ui/main.qml line 312
>  2.   change property var systemmonitorAvailableSources to property var systemmonitorAvailableSources: [] at plasma-applet-thermal-monitor/package/contents/ui/main.qml line 58
>  3.  pull #54

1. works not for me, file:///usr/share/plasma/plasmoids/org.kde.thermalMonitor/contents/ui/main.qml:312:1: Expected token `}'
2. works and fix this bug #53 :-D
3. fix this bug not

THANKS!!!

**Comment by severnaside** [comment]
@roachsinai 
> So sorry that only you reply Not Work, haven't get a Work reply(but 2 thumb up at my previous reply).

Yes, pull #54 worked for me.
Arch Linux, LTS kernel 4.19.15-1, Intel i7-7500U

**Comment by roachsinai** [comment]
@alium @severnaside glad to hear that.

**Comment by Termuellinator** [comment]
Applying pull https://github.com/kotelnik/plasma-applet-thermal-monitor/pull/54 to ~/.local/share/plasma/plasmoids/org.kde.thermalMonitor/contents/ui/main.qml (no thermalMonitor in /usr/share/ ) worked for me, too. (using manjaro)

**Comment by roachsinai** [comment]
Hi everyone, I got same effect with another plasmoid named `kargos`. And the effect is more better for me that as I use `Bumblebee`. Will show GPU temperature if I use nvidia and no GPU info completely if I'm not using GPU.

Not using Nvidia:
![sys_info](https://user-images.githubusercontent.com/9500049/51761511-3e49f500-2108-11e9-8af4-eecd812f8497.png)

Using Nvidia:
![sys_info_with_gpu](https://user-images.githubusercontent.com/9500049/51761514-4013b880-2108-11e9-967b-2f9d6b9db9a0.png)

And `kargos` makes your simple script works like a plasmoid, so I write a bash script.

What's more, `kargos` is compatible with `Bitbar` which provide lots of useful scripts. Check it at [k-argos-plugins](https://github.com/roachsinai/k-argos-plugins) and another two k-argos plugin.

**Comment by sgparry** [comment]
#54 Works for me

**Comment by GuerreroNinja** [comment]
I use kargos with hdsentinel to monitor my HDs health and temperature. Works like charm.

Pull #54 works also for me btw.

**Comment by varlesh** [comment]
DEB-package for bionic (with #54 PR):
https://launchpad.net/~varlesh-l/+archive/ubuntu/bionic/+files/plasma-applet-thermal-monitor_1.2.9git20190121_amd64.deb

**Comment by ShayBox** [comment]
1950x here using `nct`, when the temperature is unchanged it will not return, not a problem for `sensors` but the widget changes the text to off, remove the off text and keep the existing text until it changes.

**Comment by Lastique** [comment]
@varlesh That package still displays OFF for CPU temps for me in Kubuntu 19.04. I suppose, it means https://github.com/kotelnik/plasma-applet-thermal-monitor/pull/54 doesn't work.

**Comment by tairosonloa** [comment]
> @varlesh That package still displays OFF for CPU temps for me in Kubuntu 19.04. I suppose, it means #54 doesn't work.

It looks like this applet is unmaintained. #54 works, but the pull request haven't been merged. You must modify the code on your installation.

**Comment by varlesh** [comment]
@Lastique Worked, but after reboot plasma.
Added CPU temp sensor > `killall plasmashell` > **ALT+F2** > `plasmashell`
![image](https://user-images.githubusercontent.com/8083855/57724502-fd1c2780-769b-11e9-8f58-098938254a77.png)

**Comment by Lastique** [comment]
@varlesh Indeed, after a system reboot CPU temps tend to show up, although after a delay. I wasn't sure this behavior was reliable, I've only rebooted a couple of times since I installed the package. Thanks!

**Comment by fleamour** [comment]
sudo nano ~/.local/share/plasma/plasmoids/org.kde.thermalMonitor/contents/ui/main.qml

change:
`property var systemmonitorAvailableSources`
to: 
`property var systemmonitorAvailableSources: []` 

Works under openSUSE Tumbleweed.

**Comment by hellbenthorse** [comment]
> sudo nano ~/.local/share/plasma/plasmoids/org.kde.thermalMonitor/contents/ui/main.qml
> 
> change:
> `property var systemmonitorAvailableSources`
> to:
> `property var systemmonitorAvailableSources: []`
> 
> Works under openSUSE Tumbleweed.

Can confirm this works under Gentoo also, and I dare say any other Distro for that matter.

**Comment by lr222gw** [comment]
> sudo nano ~/.local/share/plasma/plasmoids/org.kde.thermalMonitor/contents/ui/main.qml
> 
> change:
> `property var systemmonitorAvailableSources`
> to:
> `property var systemmonitorAvailableSources: []`
> 
> Works under openSUSE Tumbleweed.

I just did this on Manjaro, now the widget works without having to reload it! :+1:

**Comment by lvxejay** [comment]
Is this plasma applet unmaintained? Could submit a PR if this fixed it

**Comment by msdobrescu** [comment]
Under Sabayon Linux works for kernel version 4 but not for 5, where needs manual reload.

**Comment by Rush** [comment]
I am glad I found this thread. It's a very annoying issue. Thanks for the solution.

**Comment by Robiocop** [comment]
It works for me under KDE Neon Bionic. Thanks for this tip.

**Comment by msdobrescu** [comment]
Hi, do you refer to this poor workaround as tip? It is not intended as tip, it just reveals some kernel version 4 vs version 5 difference as reason of the problem.
I'd rather drop the applet than stick with an old kernel.
Probably, the simplest workaround would be to do some refresh at the applet starting, as it manually does and works.

**Comment by fleamour** [comment]
Can anyone submit a pull request?

**Comment by roachsinai** [comment]
Hum...

In fact, I've already made that pr 7 months ago.

There is an alternative choice: [k-argos-plugins](https://github.com/roachsinai/k-argos-plugins).

And maybe the pr title is no good, I will rename it.

---

Update:

Renamed it as **Solution #53: CPU temperature goes to off state** #54 .

**Comment by Rush** [comment]
@roachsinai k-argos is not an alternative. It seems it's doing a lot more than just temperature. I don't need a Chinese calendar.

**Comment by roachsinai** [comment]
@Rush what you need is just [sysinfo.3s.sh](https://github.com/roachsinai/k-argos-plugins/blob/master/sysinfo.3s.sh), each script for a different task(like a plasmoid) with the help of *kargos* in that repo. Maybe you should first take a peek at [kargos](https://github.com/lipido/kargos). If it is not what you want , just forget it.

**Comment by msdobrescu** [comment]
Thermal monitor works for various sensors I have, I use it for CPU and video temperature and for CPU fan speed. It simply needs some update to be able to start the CPU temperature and CPU fan automatically, and possibly some more sensors, but the video temperature works from the start...
I don't see the point of discussing alternatives here.

**Comment by roachsinai** [comment]
@msdobrescu Yeah you're right man, so I made #54 when I found solution to solve this problem 7 months ago. And people always first go to issues to find whether there is a solution but not check at pull requests right? What's more, maybe another grammer chage, etc, of next kde version will break it again, and it's hard for a newbie of qml like me to find the error and solve it.

So I just share another way can do same thing or more, [sysinfo.3s.sh](https://github.com/roachsinai/k-argos-plugins/blob/master/sysinfo.3s.sh) now support Cpu temp, Gpu temp, Cpu speed, Cpu usage, memory usage. As the maintainer big @kotelnik not in active, for me more choice will be better.

**Comment by msdobrescu** [comment]
Indeed, somebody should look at the pull requests when they come, assess and accept it or reject it.
Maybe the maintainer does not work on it anymore? Or has no time? In this case, a fork might be the solution.

**Comment by roachsinai** [comment]
Don't know, @kotelnik only made an issue last year. Waiting for the return of  @kotelnik and keep going.

**Comment by afettouhi** [comment]
@roachsinai I tried out the kargos widget. I can't get it to work. I downloaded the kargos widget and downloaded the sysinfo.3s.sh from the plugin github page and set it up. But it only the widget only says starting... and nothing more

**Comment by roachsinai** [comment]
@afettouhi maybe you forgot to add run permission, if so use `chmod +x sysinfo.3s.sh`.

**Comment by afettouhi** [comment]
> @afettouhi maybe you forgot to add run permission, if so use `chmod +x sysinfo.3s.sh`.

Just tried that and still just starting... as output.

**Comment by roachsinai** [comment]
@afettouhi try to run `sysinfo.3s.sh` directly in a terminal, my output is:

```
hmank ~/./argos » ./sysinfo.3s.sh
50°2400|font=iosevka size=14 color=white
---
CPU: 5.0% |font=iosevka size=14 color=white
MEM: 67.2% |font=iosevka size=14 color=white
```

Maybe check whether you install font iosevka or change it to font you installed.

---

What's more, if still not working please make an issue in my repo to keep this thread clean.

**Comment by afettouhi** [comment]
@roachsinai this the output I get running the sh file.

```
[af@andre Hentninger]$ ./sysinfo.3s.sh
<u>34</u>°<u></u>&nbsp;<u>29</u>°|font=iosevka size=14 color=white
---
CPU: .1% |font=iosevka size=14 color=white
MEM: 12.7% |font=iosevka size=14 color=white
```

**Comment by roachsinai** [comment]
You said set it up meaning have set the executable path at kargos setting? If not make it at the kargos plasmoid settings.

If still not work please create issue at *kargos* repo or my plugin repo.  
Not suitable for discussion here.

**Comment by Zardoz89** [comment]
I make a fork over @agurenko fork with a fix to this issue.

**Comment by mariocarbone** [comment]
Is there any way to try to config it via a file? 
I removed it and reinstalled apllied the fix and don't work anyway

This is my latest config
![image](https://user-images.githubusercontent.com/14996376/82927961-c1d85800-9f81-11ea-9d14-2b294b5962b1.png)
Now i have the shitty icon :(

**Comment by gchamon** [comment]
I had ksysguard reinstalled after installing the applet and it seems to have solved the problem for me:

https://bbs.archlinux.org/viewtopic.php?pid=1977673#p1977673
